test/unittest/unittest_util_func.cpp

```diff
@@ -0,0 +1,211 @@
+
+  float x = 9871.0;
+  float sx = nntrainer::sqrtFloat(x);
+
+  if ((sx * sx - x) * (sx * sx - x) < tolerance)
+    status = ML_ERROR_NONE;
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, logFloat_01_p) {
+  int status = ML_ERROR_NONE;
+  int batch = 1;
+  int height = 1;
+  int width = 10;
+
+  nntrainer::Tensor input(batch, height, width);
+  GEN_TEST_INPUT(input, i * (width) + k + 1);
+
+  nntrainer::Tensor Results = input.apply(nntrainer::logFloat);
+
+  float *data = Results.getData();
+  ASSERT_NE(nullptr, data);
+  float *indata = input.getData();
+  ASSERT_NE(nullptr, indata);
+
+  for (int i = 0; i < batch * height * width; ++i) {
+    if ((data[i] - (float)log(indata[i])) > tolerance) {
+      status = ML_ERROR_INVALID_PARAMETER;
+      break;
+    }
+  }
+
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, sigmoid_01_p) {
+  int status = ML_ERROR_NONE;
+  int batch = 1;
+  int height = 1;
+  int width = 10;
+
+  nntrainer::Tensor input(batch, height, width);
+  GEN_TEST_INPUT(input, (k - 4) * 0.1);
+
+  nntrainer::Tensor Results = input.apply(nntrainer::sigmoid);
+
+  float *data = Results.getData();
+  ASSERT_NE(nullptr, data);
+  float *indata = input.getData();
+  ASSERT_NE(nullptr, indata);
+
+  for (int i = 0; i < batch * height * width; ++i) {
+    if ((data[i] - (float)(1 / (1 + exp(-indata[i])))) > tolerance) {
+      status = ML_ERROR_INVALID_PARAMETER;
+      break;
+    }
+  }
+
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, tanhFloat_01_p) {
+  int status = ML_ERROR_NONE;
+  int batch = 1;
+  int height = 1;
+  int width = 10;
+
+  nntrainer::Tensor input(batch, height, width);
+  GEN_TEST_INPUT(input, (k - 4) * 0.1);
+
+  nntrainer::Tensor Results = input.apply(nntrainer::tanhFloat);
+
+  float *data = Results.getData();
+  ASSERT_NE(nullptr, data);
+  float *indata = input.getData();
+  ASSERT_NE(nullptr, indata);
+
+  for (int i = 0; i < batch * height * width; ++i) {
+    if ((data[i] - (float)(tanh(indata[i]))) > tolerance) {
+      status = ML_ERROR_INVALID_PARAMETER;
+      break;
+    }
+  }
+
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, relu_01_p) {
+  int status = ML_ERROR_NONE;
+  int batch = 1;
+  int height = 1;
+  int width = 10;
+
+  nntrainer::Tensor input(batch, height, width);
+  GEN_TEST_INPUT(input, (k - 4) * 0.1);
+
+  nntrainer::Tensor Results = input.apply(nntrainer::relu);
+
+  float *data = Results.getData();
+  ASSERT_NE(nullptr, data);
+  float *indata = input.getData();
+  ASSERT_NE(nullptr, indata);
+
+  for (int i = 0; i < batch * height * width; ++i) {
+    float r = (indata[i] <= 0.0) ? 0.0 : indata[i];
+    if ((data[i] - r) > tolerance) {
+      status = ML_ERROR_INVALID_PARAMETER;
+      break;
+    }
+  }
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Main gtest
+ */
+int main(int argc, char **argv) {
+  int result = -1;
+
+  testing::InitGoogleTest(&argc, argv);
```

Comment:
Add to handle exceptions thrown by GTest API.
Refer this PR : https://github.com/nnstreamer/nnstreamer/pull/2077

Reply:
Thanks. Fixed.